trace-utils/src/span/v05/mod.rs

```diff
@@ -29,24 +32,146 @@ pub struct Span {
     pub r#type: u32,
 }
 
+///This structure is a wrapper around a slice of span events
+///
+/// It is meant to overrdide the default serialization, so we can serialize attributes
+/// differently from the original impl.
+/// Span events are serialized to JSON and added to "meta" when serializing to v0.5
+///
+/// The main difference with messagepacck serialization is that attributes with any types
+/// are supposed to be mapped to their natural JSON representation.
+///
+/// Sadly, I haven't found a good way of overriding the default Serialize behavior, other
+/// than just doing it for the whole data structures that embed it.
```

Comment:
```suggestion
/// This structure is a wrapper around a slice of span events
///
/// It is meant to override the default serialization, so we can serialize attributes
/// differently from the original impl.
/// Span events are serialized to JSON and added to "meta" when serializing to v0.5
///
/// The main difference with messagepack serialization is that attributes with any types
/// are supposed to be mapped to their natural JSON representation.
```

you can skip the justification and personalization.